### PR TITLE
Remove deprecated JavaExec.(get|set)Main

### PR DESCRIPTION
**Pull request by rieske**
https://github.com/gradle/gradle/issues/17151

### CODE REVIEW

**Comment by tresat** [comment]
@bot-gradle test QFL

**Comment by tresat** [comment]
@bot-gradle test QFL

**Comment by tresat** [comment]
@bot-gradle test QFL

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
Link to potential fix for asciidoctor (plugin version will need to be updated in this branch) - https://github.com/asciidoctor/asciidoctor-gradle-plugin/pull/642

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Comment by tresat** [comment]
@bot-gradle test this

**Review by jvandort — APPROVED** [review]

File: subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java [hunk 1]
```diff
@@ -78,7 +65,16 @@ public interface JavaExecSpec extends JavaForkOptions, BaseExecSpec {
      */
```
Comment (big-guy) [inline subprojects/core-api/src/main/java/org/gradle/process/JavaExecSpec.java:65]: The docs here still say "Gradle 8.0"

**Review by big-guy — CHANGES_REQUESTED** [review]

**Review by big-guy — APPROVED** [review]

**Comment by tresat** [comment]
@bot-gradle test and merge

**Pull request merged**